### PR TITLE
Use `IsList(toList,fromList)` instead of specialised functions

### PR DESCRIPTION
**Pull request by carbolymer**
# Changelog

```yaml
- description: |
    Use IsList(toList,fromList) instead of specialised functions.
    Use `Pretty Coin` instance for printing lovelace amounts.
# uncomment types applicable to the change:
  type:
  # - feature        # introduces a new feature
  # - breaking       # the API has changed in a breaking way
  # - compatible     # the API has changed but is non-breaking
  # - optimisation   # measurable performance improvements
   - refactoring    # QoL changes
  # - bugfix         # fixes a defect
  # - test           # fixes/modifies tests
  # - maintenance    # not directly related to the code
  # - release        # related to a new release preparation
  # - documentation  # change in code docs, haddocks...
```

# Context

na

# Checklist

- [ ] Commit sequence broadly makes sense and commits have useful messages
- [ ] New tests are added if needed and existing tests are updated. See [Running tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- [ ] Self-reviewed the diff

### CODE REVIEW

**Review by Jimbo4350 — APPROVED** [review]

File: cardano-cli/src/Cardano/CLI/Json/Friendly.hs [hunk 1]
```diff
@@ -5,6 +5,9 @@
 {-# LANGUAGE ScopedTypeVariables #-}
 {-# LANGUAGE TypeApplications #-}
 {-# LANGUAGE TypeOperators #-}
+{-# OPTIONS_GHC -Wno-unrecognised-pragmas #-}
```
Comment (smelc) [inline cardano-cli/src/Cardano/CLI/Json/Friendly.hs:8]: Why is that required?

Comment (carbolymer) [reply cardano-cli/src/Cardano/CLI/Json/Friendly.hs:8]: https://github.com/IntersectMBO/cardano-cli/pull/856#discussion_r1716522158
